src/Microsoft.Identity.Web/AuthorityHelpers.cs

```diff
@@ -34,7 +34,7 @@ internal static string EnsureAuthorityIsV2(string authority)
             return authority;
         }
 
-        internal static string? BuildCiamAuthorityIfNeeded(string? authority)
+        internal static string? BuildCiamAuthorityIfNeeded(string? authority, out bool preserveAuthority)
```

Comment:
In the spirit of "Single responsability principle" (the S in SOLID), I would rename this new arg as "isCiamCudAuthority", i.e. the authority helper BuildCiamAuthorityIfNeeded should not need to know about concepts like "preseve authority".

Reply:
@jennyf19 do you have any thoughts on renaming this method?

Reply:
But if we wanted to, BuildCiamOrOidcAuthority is ok.

Reply:
I am leaning more towards what @jennyf19 said. This method will only perform an action if the authority contains a Ciam suffix. but if it does not, then it does not build anything really. thus, the if needed is more like, "if we need to build then do it, otherwise, it does not."